includes/class-wc-stripe-payment-method-configurations.php

```diff
@@ -58,29 +58,71 @@ class WC_Stripe_Payment_Method_Configurations {
 	 * @return object|null
 	 */
 	private static function get_primary_configuration( $force_refresh = false ) {
-		if ( ! $force_refresh ) {
+		// Only allow fetching payment configuration once per minute.
+		$fetch_cooldown = defined( 'IS_TESTING' ) ? 0 : get_option( 'wcstripe_payment_method_config_fetch_cooldown', 0 );
+		$is_in_cooldown = $fetch_cooldown > time();
+		if ( ! $force_refresh || $is_in_cooldown ) {
 			$cached_primary_configuration = self::get_payment_method_configuration_from_cache();
 			if ( $cached_primary_configuration ) {
 				return $cached_primary_configuration;
 			}
+
+			// If we are hitting the API too much, and our main cache is not working, use the fallback cache.
+			if ( $is_in_cooldown ) {
+				$fallback_cache = self::get_payment_method_configuration_from_cache( true );
+				if ( $fallback_cache ) {
+					return $fallback_cache;
+				}
+
+				// If even the fallback cache is not available, return minimal card configuration.
+				return self::get_minimal_card_config();
+			}
```

Comment:
I updated this to wrap the fallback logic in `if ( $is_in_cooldown )` because,

1. frontend may trigger call to API, i.e. `$force_refresh` is false
2. transient may already be expired, i.e. `$cached_primary_configuration` is empty
3. if we are not in cooldown, i.e. hitting the API too much, we want to retrieve a fresh PMC and not use stale fallbacks